### PR TITLE
Feat/galxe

### LINKED ISSUE #21: Look into Galxe

### PR DESCRIPTION
**Pull request by Th0rgal**
# Add New `/galxe/verify` POST Endpoint

This PR introduces a new endpoint `/galxe/verify` for verifying sales linked to an email address. It is intended to be used  by the Galxe backend.

## Usage:

Make a POST request with a JSON payload containing the email you wish to verify:

```json
{
    "email": "your_email_here@domain.com"
}
```

The endpoint will return:

- `"0x00000000000000000000000000000001"` if a sale is found with the specified email.
- `"0x00000000000000000000000000000000"` otherwise.

### 1. Verifying a known email:

```json
    {
        "email": "thomas@starknet.id"
    }
```

#### Expected response:

```json
    {
        "result": "0x00000000000000000000000000000001"
    }
```

### 2. Verifying an unknown email:

```json
    {
        "email": "notfound@starknet.id"
    }
```

#### Expected response:

```json
    {
        "result": "0x00000000000000000000000000000000"
    }
```

closes #21

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -14,7 +14,17 @@ macro_rules! pub_struct {
 
 pub_struct!(Server { port: u16, });
 
-pub_struct!(Database {
+pub_struct!(Databases {
+    starknetid: StarknetidDb,
+    sales: SalesDb,
+});
+
+pub_struct!(StarknetidDb {
```
Comment (irisdv) [inline src/config.rs:22]: Maybe we could have a common struct `Database` as the data is the same for both db

File: src/main.rs [hunk 2]
```diff
@@ -18,18 +18,27 @@ use tower_http::cors::{Any, CorsLayer};
...
+            .unwrap()
+            .database(&conf.databases.starknetid.name),
+        sales_db: Client::with_options(sales_client_options)
             .unwrap()
-            .database(&conf.database.name),
+            .database(&conf.databases.sales.name),
     });
     if shared_state
-        .db
+        .starknetid_db
```
Comment (irisdv) [inline src/main.rs:41]: We don't need to ping the sales db as well ?

**Review by irisdv — APPROVED** [review]
lgtm! I just left 2 small comments in the code

File: src/config.rs (hunk 1 above)
Comment (Th0rgal) [reply src/config.rs:22]: good idea

File: src/main.rs (hunk 2 above)
Comment (Th0rgal) [reply src/main.rs:41]: let's do it too

**Pull request merged**